### ISSUE TITLE
hide mpd module completely info when stopped or disconnected

### ISSUE THREAD

**Issue by DyslexicAtheist** [Alexays/Waybar#782]
Hi,

as the title says I'm trying to not show the mpd info when it's in state `stopped` or `disconnected`. My current setup achieves this using the following empty format options in `waybar/config`:

```
        "format-disconnected": "",
        "format-stopped": "",
```

this is done in combination with transparent `color and background-color in style.css:

```
#mpd {
    padding: 0 10px;
    margin: 0 4px;
    background-color: #475f94;
    color: #f3f4f5;
}

#mpd.disconnected {
    background-color: transparent;
    color: transparent;
}

#mpd.stopped {
    background-color: transparent;
    color: transparent;
}

#mpd.paused {
    background-color: #51a37a;
}
```

is there another way to do this? Apologies for the somewhat silly question, but it felt like a bit of a hack doing this, so I'm unsure if this is the correct method. I couldn't find any info in the waybar-mpd (5) that instructs me how to achieve this. (I thought this would be a fairly standard way of how people want to set this up but maybe it's just me ;))

Thanks for any pointers.

my version is:
Waybar v0.9.2-140-g759602a (branch 'master')

thanks,
~DA

**Comment by fauxmight** [comment]
I strongly agree with your point that implementing something like the "exec-if" field from the custom module for **other** modules would be tremendously useful.

In general, conditionally displaying/executing modules as implemented in #394 seems like an extremely useful capability, though, according to the pull request, this idea has been rejected by waybar's creator in favor of a different (also abandoned) concept. (#100).

~Non-dyslexic Messianic Jew who mirror-writes anyway
[ ^ Just toying with your handle @DyslexicAtheist ; no disrespect intended]

**Comment by awused** [comment]
Seems like even this workaround broke at some point as the `.stopped` class is never applied at git head.

**Comment by Alexays** [comment]
There's a cleaner way — you don't need the transparent-color CSS hack. When a module's format string is empty, Waybar hides the label entirely (see `src/modules/mpd/mpd.cpp`, which calls `label_.hide()` when the format is empty). So just set:

```jsonc
"mpd": {
    "format-stopped": "",
    "format-disconnected": ""
}
```

and drop the `#mpd.stopped`/`#mpd.disconnected` transparency rules — the module collapses completely (no leftover padding/margin) in those states. Closing, but feel free to reopen if it misbehaves. 🙏
